src/main/java/hudson/remoting/Launcher.java

```diff
@@ -550,10 +683,6 @@ public List<String> parseJnlpArguments() throws ParserConfigurationException, SA
                 List<String> jnlpArgs = new ArrayList<>();
                 for( int i=0; i<argElements.getLength(); i++ )
                         jnlpArgs.add(argElements.item(i).getTextContent());
-                if (agentJnlpCredentials != null) {
-                    jnlpArgs.add("-credentials");
-                    jnlpArgs.add(agentJnlpCredentials);
-                }
```

Comment:
This code is actually being kept, just moved next to its siblings for legibility rather than in this random place.